playground/legacy/__tests__/legacy.spec.ts

```diff
@@ -52,22 +52,22 @@ test('generates assets', async () => {
     () => page.textContent('#assets'),
     isBuild
       ? [
-          'index: 404',
-          'index-legacy: 404',
-          'chunk-async: 404',
-          'chunk-async-legacy: 404',
-          'immutable-chunk: 200',
-          'immutable-chunk-legacy: 200',
-          'polyfills-legacy: 404',
+          'index: text/html; charset=utf-8',
+          'index-legacy: text/html; charset=utf-8',
+          'chunk-async: text/html; charset=utf-8',
+          'chunk-async-legacy: text/html; charset=utf-8',
+          'immutable-chunk: application/javascript',
+          'immutable-chunk-legacy: application/javascript',
+          'polyfills-legacy: text/html; charset=utf-8',
```

Comment:
Should we leave them 404 in `appType: 'mpa'`? It would be odd if they fallback to the `index.html` 🤔

Reply:
Legacy playground is not `appType: 'mpa'`. This part is the result of fetching all chunks.
https://github.com/vitejs/vite/pull/14142/files/79491351b244ee9ceff9cccbacb810fe47744b28#diff-b17ad79ba0d84b7069324d7eb460e718df66793c5cc903f5b6ff92b5adb73c20

Reply:
Yeah the test here is fine since it's an spa, but for MPAs (which I manually tested) it also has this same result, and I wonder if that's correct. For example, if we fetch `index` (which doesn't exist), in SPA it should fallback to `/index.html`, but in MPA, it should 404 instead?

Reply:
Ah, I see. That behavior was there before this PR.
This PR just makes the paths with `.` work like the paths without `.`.

| Access | Response before this PR | Response after this PR |
|---------|---------------------------|---------------------------|
| `/` | `/index.html` | `/index.html` |
| `/foo` | `/index.html` | `/index.html` |
| `/foo/` | 404 | 404 |
| `/foo.png` | 404 | `/index.html` |
| `/nested` | `/index.html` | `/index.html` |
| `/nested/` | `/nested/index.html` | `/nested/index.html` |
| `/nested/foo` | `/index.html` | `/index.html` |
| `/nested/foo.png` | 404 | `/index.html` |

(in MPA)